https://bugs.gentoo.org/887769
https://gitlab.gnome.org/GNOME/gthumb/-/merge_requests/42

From 2133520496afd179702953a5d07aa99b5209f9cf Mon Sep 17 00:00:00 2001
From: Chris Mayo <aklhfex@gmail.com>
Date: Wed, 21 Dec 2022 19:21:12 +0000
Subject: [PATCH] raw: make compatible with LibRaw 0.21.0 release

Memory error callback removed
https://github.com/LibRaw/LibRaw/blob/979160ff13/Changelog.txt#L127

Typo in LIBRAW_OPTIONS_* flag names fixed
https://github.com/LibRaw/LibRaw/blob/979160ff13/Changelog.txt#L170
--- a/extensions/raw_files/gth-metadata-provider-raw.c
+++ b/extensions/raw_files/gth-metadata-provider-raw.c
@@ -62,7 +62,11 @@ gth_metadata_provider_raw_read (GthMetadataProvider *self,
 	if (!_g_mime_type_is_raw (gth_file_data_get_mime_type (file_data)))
 		return;
 
+#if LIBRAW_COMPILE_CHECK_VERSION_NOTLESS(0, 21)
+	raw_data = libraw_init (LIBRAW_OPTIONS_NO_DATAERR_CALLBACK);
+#else
 	raw_data = libraw_init (LIBRAW_OPIONS_NO_MEMERR_CALLBACK | LIBRAW_OPIONS_NO_DATAERR_CALLBACK);
+#endif
 	if (raw_data == NULL)
 		goto fatal_error;
 
--- a/extensions/raw_files/main.c
+++ b/extensions/raw_files/main.c
@@ -213,7 +213,11 @@ _cairo_image_surface_create_from_raw (GInputStream  *istream,
 	size_t         size;
 	GthImage      *image = NULL;
 
+#if LIBRAW_COMPILE_CHECK_VERSION_NOTLESS(0, 21)
+	raw_data = libraw_init (LIBRAW_OPTIONS_NO_DATAERR_CALLBACK);
+#else
 	raw_data = libraw_init (LIBRAW_OPIONS_NO_MEMERR_CALLBACK | LIBRAW_OPIONS_NO_DATAERR_CALLBACK);
+#endif
 	if (raw_data == NULL) {
 		_libraw_set_gerror (error, errno);
 		goto fatal_error;
@@ -300,7 +304,11 @@ _cairo_image_surface_create_from_raw (GInputStream  *istream,
 		if ((original_width != NULL) && (original_height != NULL)) {
 			libraw_close (raw_data);
 
+#if LIBRAW_COMPILE_CHECK_VERSION_NOTLESS(0, 21)
+			raw_data = libraw_init (LIBRAW_OPTIONS_NO_DATAERR_CALLBACK);
+#else
 			raw_data = libraw_init (LIBRAW_OPIONS_NO_MEMERR_CALLBACK | LIBRAW_OPIONS_NO_DATAERR_CALLBACK);
+#endif
 			if (raw_data == NULL)
 				goto fatal_error;
 
-- 
GitLab
